Linux bugfixes for globcasedetect

JohnoKing opened the pull request:
src/lib/libast/features/lib,
src/lib/libast/path/pathicase.c:
* FAT32/VFAT file systems on Linux don't support `FS_CASEFOLD_FL`, which caused case-insensitive globbing to break. This was fixed by checking for FAT attributes with ioctl, then checking for `FS_CASEFOLD_FL` if that fails. Reproducer using a UEFI boot partition:  
```sh
$ set --globcasedetect
$ echo /boot/eF*
/boot/eF*
```
* The check for `FS_CASEFOLD_FL` didn't work correctly; I still wasn't able to get `--globcasedetect` to work on a case-insensitive ext4 folder. Fix that by adding missing parentheses (with this change it's now working fine on my system).

src/cmd/ksh93/features/options:
* Prefer `/boot/EFI` and `/boot` over `/` since those could be FAT partitions. This should allow ksh to detect `SHOPT_GLOBCASEDET` support on Linux kernels older than 5.2.

JohnoKing committed:
Linux bugfixes for globcasedetect (re: 71934570)

src/lib/libast/features/lib,
src/lib/libast/path/pathicase.c:
- FAT32 file systems on Linux don't support FS_CASEFOLD_FL, which
  caused globbing to break. Reproducer using a UEFI boot partition:
      $ echo /boot/eF*
      /boot/eF*
  This is fixed by checking for FAT attributes with ioctl, then
  checking for FS_CASEFOLD_FL if that fails.
- The check for FS_CASEFOLD_FL didn't work correctly; I still wasn't
  able to get --globcasedetect to work on a case-insensitive ext4
  folder. Fix that by adding missing parentheses (with this change
  it's now working fine on my system).

src/cmd/ksh93/features/options:
- Prefer /boot/EFI and /boot over / since those may be VFAT partitions.
  This should allow ksh to detect SHOPT_GLOBCASEDET support on Linux
  kernels older than 5.2.

McDutchie commented on src/cmd/ksh93/features/options:
```diff
@@ -62,8 +62,17 @@ tst	note{ can we probe file system case insensitivity }end output{
+		char *dir;
 		int r;
-		r = pathicase("/");
+		/* Prioritize /boot/EFI and /boot since those could
```
Are these additions actually necessary? Note the feature test only returns false if `pathicase` returns -1 **and** sets `errno` to `ENOSYS`. According to the man page I found on the web, `ioctl`(2) on Linux does not ever set `errno` to `ENOSYS`.

JohnoKing replied:
I just tested it and it turns out that change was unnecessary. Now reverted.

JohnoKing committed:
features/options: Revert unnecessary change

As it turns out older Linux (tested using 5.1) doesn't need
this change. Now reverted.

McDutchie committed:
Minor cosmetic tweaks

Note that the libast code (unlike the ksh93 code) generally has a
space between 'if'/'while' and the opening parenthesis.
